src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -115,6 +116,7 @@ export interface DiscoverSidebarResponsiveProps {
  */
 export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps) {
   const services = useDiscoverServices();
+  const { isPlainRecord } = useDiscoverLayoutContext();
```

Comment:
Same here, having isPlainRecord as a helper using  documentState on L134 would IMO be a bit simpler